### DISCUSSION TITLE
how to display query result from supabase to flutter?

### DISCUSSION THREAD

**Discussion by irsanachmad21** [supabase/supabase#31660] [Questions]
i have some query in supabase like this:
-- BY YEAR
SELECT a.category, COUNT(*) AS amount
FROM tb_activities a
WHERE a.year_id = 2024 -- Change Year Here
GROUP BY a.category
ORDER BY amount DESC;

-- BY MONTH AND YEAR
SELECT a.category, COUNT(*) AS amount
FROM tb_activities a
JOIN tb_months m ON a.month_id = m.month_id
WHERE a.year_id = 2024 --Change Year Here
AND m.month = 'November' --Change Month Here
GROUP BY a.category
ORDER BY amount DESC;

when i run the query above directly from SQL Editor in supabase the result appears but i want the result to appear in the mobile application that i created. how is the code to display it?

sorry if my english is confusing

**Comment by silentworks** [comment]
You write the query as a Postgres function and then call it with the flutter SDK using the `.rpc` method. https://supabase.com/docs/reference/dart/rpc
